### PR TITLE
fix(elasticsearch sink): Revert add `api_version` option

### PR DESCRIPTION
**Pull request by jszwedko**
Reverts vectordotdev/vector#14918

This is causing soak tests to fail because lading doesn't respond correctly to the version endpoint. This fact also highlights that the PR being reverted could be considered a "breaking change" so, before we re-introduce it, we should decide to either preserve compatibility (default to `v7`) or make the breaking change and add an upgrade note to the next release so that users know to expect it.

cc/ @ktff

### CODE REVIEW

**Comment by jszwedko** [comment]
We could have rolled forward here with whatever we decide but given it's the end of the week I think reverting until we can decide the desired behavior is better.

**Comment by jszwedko** [comment]
Noticing also that the error message when autodiscovery fails due to deserialization errors isn't great:

```
  2022-10-28T22:01:03.636082Z ERROR vector::topology: Configuration error. error=Sink "elasticsearch": missing field `version` at line 1 column 123
```

It makes it look like the config file is missing an option rather than that the autodiscovery failed. We should fix that too.

**Review by fuchsnj — APPROVED** [review]

**Comment by jszwedko** [comment]
Merging to unblock master since comparison soaks ran.

**Comment by ktff** [comment]
Missed that soak tests were doing something like this. I'm assuming lading is not spinning up actual Elastisearch server but just pretending to be one. That's the same issue with `sematext` and `axiom` sinks that wrap `elastisearch` sink. Their servers only partially support Elastcisearch API so their `api_version` needed to be set to specific version, so we can also set `api_version` for soak tests. I'm also wondering how soak tests passed in the PR in the first place?

Regarding error message: 
>  2022-10-28T22:01:03.636082Z ERROR vector::topology: Configuration error. error=Sink "elasticsearch": missing field `version` at line 1 column 123

true that should be nicer.

Regarding breaking change, we can make failure to auto determine version non fatal and just log a warning for now. Then in a later version this can be turned into a hard error. In that way all of the current configs would continue to work but users would see a warning if the sink failed to auto determine version which would help to make the transition smoother.

**Comment by jszwedko** [comment]
> Missed that soak tests were doing something like this. I'm assuming lading is not spinning up actual Elastisearch server but just pretending to be one. That's the same issue with `sematext` and `axiom` sinks that wrap `elastisearch` sink. Their servers only partially support Elastcisearch API so their `api_version` needed to be set to specific version, so we can also set `api_version` for soak tests. I'm also wondering how soak tests passed in the PR in the first place?

Yeah, that's right, lading wasn't issuing the expected response. The soaks actually failed on the PR but through a quirk of Github Actions behavior with "required checks" it didn't block merging because it thought it was "skipped". I'd like to plug that hole at some point.

> Regarding error message:
> 
> > 2022-10-28T22:01:03.636082Z ERROR vector::topology: Configuration error. error=Sink "elasticsearch": missing field `version` at line 1 column 123
> 
> true that should be nicer.

👍 

> Regarding breaking change, we can make failure to auto determine version non fatal and just log a warning for now. Then in a later version this can be turned into a hard error. In that way all of the current configs would continue to work but users would see a warning if the sink failed to auto determine version which would help to make the transition smoother.

👍 I like that, falling back to the current default of 7 (or 8 if they have `supress_type: false`).

**Comment by ktff** [comment]
@jszwedko 

> >  Regarding breaking change, we can make failure to auto determine version non fatal and just log a warning for now. Then in a later version this can be turned into a hard error. In that way all of the current configs would continue to work but users would see a warning if the sink failed to auto determine version which would help to make the transition smoother.
>
> +1 I like that, falling back to the current default of 7 (or 8 if they have supress_type: false).

Not sure if there is a misunderstanding somewhere. 

Currently by default `suppress_type_name: false` which would make current assumed version 6 or 7. But for 7 we can set this to true so to avoid getting deprecation warnings from Elastisearch. While for 8 we need to set this to true to be compatible.

Going by this I would set default as 6 and let `suppress_type_name` overwrite it if set.

**Pull request merged**
